plugins/cpp_metrics/service/src/cppmetricsservice.cpp

```diff
@@ -81,6 +96,111 @@ double CppMetricsServiceHandler::getSingleCppMetricForAstNode(
+
+  _transaction([&, this](){
+    typedef odb::query<model::CppFileMetrics> CppModuleMetricsQuery;
+
+    auto moduleMetrics = _db->query<model::CppFileMetrics>(
+      CppModuleMetricsQuery::file == std::stoull(fileId_));
+
+    for (const auto& moduleMetric : moduleMetrics)
+    {
+      metric.type = static_cast<CppModuleMetricsType::type>(moduleMetric.type);
+      metric.value = moduleMetric.value;
+      _return.push_back(metric);
+    }
+  });
+}
+
+void CppMetricsServiceHandler::getCppAstNodeMetricsForPath(
+  std::vector<CppAllMetricsAstNode>& _return,
+  const std::string& path_)
+{
+  _transaction([&, this]()
+  {
+    typedef odb::query<model::CppAstNodeMetrics> CppAstNodeMetricsQuery;
+    typedef odb::result<model::CppAstNodeMetrics> CppAstNodeMetricsResult;
+    typedef odb::query<model::CppAstNodeFilePath> CppAstNodeFilePathQuery;
+    typedef odb::result<model::CppAstNodeFilePath> CppAstNodeFilePathResult;
+
+    CppAstNodeFilePathResult nodes = _db->query<model::CppAstNodeFilePath>(
+      CppAstNodeFilePathQuery::LocFile::path.like(path_ + '%'));
+
+    if (nodes.empty())
+    {
+      core::InvalidInput ex;
+      ex.__set_msg("Invalid metric type for path: " + path_);
+      throw ex;
+    }
+
+    for (const auto& node : nodes)
+    {
+      auto metricsQuery = _db->query<model::CppAstNodeMetrics>(
+        CppAstNodeMetricsQuery::astNodeId == node.id);
+      std::vector<CppMetricsAstNode> metrics;
+
+      CppMetricsAstNode metricsAstNode;
+      for (const auto& metric : metricsQuery)
+      {
+        metricsAstNode.type = static_cast<CppAstNodeMetricsType::type>(metric.type);
+        metricsAstNode.value = metric.value;
+      }
+
+      CppAllMetricsAstNode nodeMetric;
+      nodeMetric.id = std::to_string(node.id);
+      nodeMetric.metrics = metrics;
+      _return.push_back(nodeMetric);
+    }
+  });
+}
+
+void CppMetricsServiceHandler::getCppFileMetricsForPath(
+  std::vector<CppAllMetricsModule>& _return,
+  const std::string& path_)
+{
+  _transaction([&, this]()
+  {
+    typedef odb::query<model::File> FileQuery;
+    typedef odb::result<model::File> FileResult;
+    typedef odb::query<model::CppFileMetrics> CppFileMetricsQuery;
+    typedef odb::result<model::CppFileMetrics> CppFileMetricsResult;
+
+    FileResult descendants = _db->query<model::File>(
+      FileQuery::path.like(path_ + '%'));
+
+    if (descendants.empty())
+    {
+      core::InvalidInput ex;
+      ex.__set_msg("Invalid metric type for path: " + path_);
+      throw ex;
+    }
+
+    for (const auto& file : descendants)
+    {
+      CppFileMetricsResult metricsQuery = _db->query<model::CppFileMetrics>(
+        CppFileMetricsQuery::file == file.id);
+      std::vector<CppMetricsModule> metrics;
+
+      CppMetricsModule metricsModule;
+      for (const auto& metric : metricsQuery)
+      {
+        metricsModule.type = static_cast<CppModuleMetricsType::type>(metric.type);
+        metricsModule.value = metric.value;
+      }
```

Comment:
A new `metricsModule` is computed, but not added to the `metrics` vector, therefore the result is always empty.

Reply:
Fixed this in 0936a195ad87688138a03977d8fcafa5ab62eef5.